apps/site/assets/ts/components/MatchHighlight.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { DetailedReactHTMLElement, useEffect, useState } from "react";
+
+// React.createElement is used as opposed to the dangerouslySetInnerHTML
+// prop to make the component contents easier to parse in testing
+
+interface Props {
+  text: string;
+  query: string;
+}
+
+// React component for highlighting matching text
+// will wrap the first substring matching the query
+// in a <b></b> and return the HTML result
+const MatchHighlight = ({
+  text,
+  query
+}: Props): DetailedReactHTMLElement<{}, HTMLElement> => {
+  const [childNodes, setChildNodes] = useState<
+    (DetailedReactHTMLElement<{}, HTMLElement> | string)[]
+  >([text]);
+
+  useEffect(
+    () => {
+      if (query !== "") {
+        const q = new RegExp(query, "i");
+        const matched = text.match(q);
+        if (matched && matched[0]) {
+          // split text on first instance of match only
+          const [precedingTextNode, ...moreTextNodes] = text.split(matched[0]);
```

Comment:
Ah, I think I had been approaching this from the standpoint of, well, they become text nodes in the end. Also `moreTexts` (I'd wanted to keep it plural as it could be of any length) sounded weird to me.

Reply:
`moreTexts` sounds fine to me, but so would something generic like `rest` or `others`; this variable is only relevant for the very next line anyway.